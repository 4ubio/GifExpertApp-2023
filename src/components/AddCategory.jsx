import { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) => { //Event es el cambio que se esta efectuando dentro del input. Desestructuramos y obtenemos target
        setInputValue(target.value); //Actualizamos el valor
    }

    const onSubmit = (event) => {
        event.preventDefault(); //Evitar recargar el formulario al dar enter
        if (inputValue.trim().length <= 1) return; //Evitar inputs vacios
        onNewCategory(inputValue.trim()); //Agregar el input nuevo a las categorias
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit} aria-label='form'>
            <input 
                type="text"
                placeholder='Search Gifs'
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}