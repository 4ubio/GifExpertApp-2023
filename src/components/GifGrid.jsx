import { useEffect, useState } from 'react'
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {

    const {images, isLoading} = useFetchGifs(category); //Nuestro custom hook

    return (
        <>
           <div key={category}>

                <h3 className='title animate__animated animate__bounceIn'>{category}</h3>
                {isLoading && (<h2 className='animate__animated animate__bounceIn'>Loading...⌛️</h2>)}

                <div className='card-grid'>
                    {images.map( (image) => (
                        <GifItem 
                            key={image.id}
                            {...image} //Para esparcir todas las propiedades dentro de nuestro componente
                        />
                    ))}
                </div>
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}