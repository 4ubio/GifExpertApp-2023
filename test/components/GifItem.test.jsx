import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Test on GifItem', () => {

    const title = 'Pokemon';
    const url = 'https://pokemon.com/pikachu.png';

    test('should match with snapshot', () => {
        const {container} = render(<GifItem title={title} url = {url}/>);
        expect(container).toMatchSnapshot();
    });

    test('should display img with the correct url and alt', () => { 
        render(<GifItem title={title} url={url} />);
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)

        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)
    })

    test('should display the title in the component', () => { 
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    })
})