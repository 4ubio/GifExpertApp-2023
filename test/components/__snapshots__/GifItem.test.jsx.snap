// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on GifItem should match with snapshot 1`] = `
<div>
  <div
    class="card animate__animated animate__bounceIn"
  >
    <img
      alt="Pokemon"
      src="https://pokemon.com/pikachu.png"
    />
    <p>
      Pokemon
    </p>
  </div>
</div>
`;
